### PR TITLE
[Sprint Maintain] Create New Password When Forgot password

### PR DESCRIPTION
**Pull request by Tuanlvt**
Create New Password When Forgot password
link redmine: https://dev.framgia.com/redmine/issues/131200

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/createnewpassword/CreateNewPasswordPresenter.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Listens to user actions from the UI ({@link CreateNewPasswordActivity}), retrieves the data and
+ * updates
+ * the UI as required.
+ */
+final class CreateNewPasswordPresenter implements CreateNewPasswordContract.Presenter {
+    private static final String TAG = CreateNewPasswordPresenter.class.getName();
+
+    private CreateNewPasswordContract.ViewModel mViewModel;
+
+     CreateNewPasswordPresenter() {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/createnewpassword/CreateNewPasswordPresenter.java]: ERROR: 'ctor def modifier' have incorrect indentation level 5, expected level should be 4.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -332,5 +332,7 @@
     <string name="the_number_of_rest_days_in_year">The number of rest days(in year)</string>
     <string name="number_of_days_off_added_in_this_year">Number of days off added in this year</string>
     <string name="total_of_days_off_remaining_in_year">Total of days off remaining in year</string>
+    <string name="please_enter_a_new_password">Please enter a new password</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`please_enter_a_new_password`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -332,5 +332,7 @@
...
+    <string name="reset_password">Reset Password</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`reset_password`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.please_enter_a_new_password` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values-vi/strings.xml [hunk 4]
```diff
@@ -319,5 +319,6 @@
     <string name="the_number_of_rest_days_in_year">Số ngày nghỉ còn lại trong năm</string>
     <string name="number_of_days_off_added_in_this_year">Số ngày nghỉ được thêm trong năm hiện tại</string>
     <string name="total_of_days_off_remaining_in_year">Tổng số ngày nghỉ còn lại trong năm</string>
-
+    <string name="please_enter_a_new_password">Vui lòng nhập vào mật khẩu mới</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values-vi/strings.xml]: WARNING: [UnusedResources] The resource `R.string.please_enter_a_new_password` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.reset_password` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values-vi/strings.xml [hunk 5]
```diff
@@ -319,5 +319,6 @@
...
+    <string name="reset_password">Đặt lại mật khẩu</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values-vi/strings.xml]: WARNING: [UnusedResources] The resource `R.string.reset_password` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/com/framgia/wsm/screen/login/LoginActivity.java [hunk 6]
```diff
@@ -30,6 +35,12 @@ protected void onCreate(Bundle savedInstanceState) {
                 DataBindingUtil.setContentView(this, R.layout.activity_login);
 
         binding.setViewModel((LoginViewModel) mViewModel);
+
+        Intent intent = getIntent();
+        Uri urlResetPassword = intent.getData();
+        String tokenResetPassword;
+        tokenResetPassword = String.valueOf(urlResetPassword).replace(URI_RESET_PASSWORD, "");
+        Log.d("Token Reset Password", tokenResetPassword);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginActivity.java]: pls remove log

File: app/src/main/java/com/framgia/wsm/screen/login/LoginActivity.java [hunk 7]
```diff
@@ -1,8 +1,11 @@
 package com.framgia.wsm.screen.login;
 
+import android.content.Intent;
 import android.databinding.DataBindingUtil;
+import android.net.Uri;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginActivity.java]: ERROR: Unused import: android.util.Log. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.util.Log'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/wsm/screen/login/LoginActivity.java [hunk 8]
```diff
@@ -30,6 +34,11 @@ protected void onCreate(Bundle savedInstanceState) {
                 DataBindingUtil.setContentView(this, R.layout.activity_login);
 
         binding.setViewModel((LoginViewModel) mViewModel);
+
+        Intent intent = getIntent();
+        Uri urlResetPassword = intent.getData();
+        String tokenResetPassword;
+        tokenResetPassword = String.valueOf(urlResetPassword).replace(URI_RESET_PASSWORD, "");
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginActivity.java]: pls check !=null

**Pull request merged**